source/common/ssl/context_config_impl.cc

```diff
@@ -308,13 +308,12 @@ ServerContextConfigImpl::ServerContextConfigImpl(
 
         return ret;
       }()) {
-  // TODO(PiotrSikora): Support multiple TLS certificates.
   if ((config.common_tls_context().tls_certificates().size() +
        config.common_tls_context().tls_certificate_sds_secret_configs().size()) == 0) {
     throw EnvoyException("No TLS certificates found for server context");
-  } else if ((config.common_tls_context().tls_certificates().size() +
-              config.common_tls_context().tls_certificate_sds_secret_configs().size()) > 1) {
```

Comment:
I think you may have multiple sds config for single cluster/listener? https://github.com/envoyproxy/envoy/blob/master/api/envoy/api/v2/auth/cert.proto#L239

Reply:
But it doesn't work with multiple certificates, because callback clears all other certificates on update, see:
https://github.com/envoyproxy/envoy/blob/8475fcf057aa1ad1978ddcc823834249efca37fc/source/common/ssl/context_config_impl.cc#L186